### PR TITLE
[EI-909] Add "buildMessage" attribute to management console

### PR DESCRIPTION
**Pull request by Bhashinee**
Fix the issue of not supporting "buildMessage" attribute of failover and loadbalance endpoint in the management console.
Related issue : https://github.com/wso2/product-ei/issues/909

### CODE REVIEW

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java [hunk 1]
```diff
@@ -52,6 +53,14 @@ public void setProperties(String properties) {
         this.properties = properties;
     }
 
+    public boolean isBuildMessage(){
```
Comment (Manuri) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java]: Fix formatting

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java [hunk 2]
```diff
@@ -52,6 +53,14 @@ public void setProperties(String properties) {
...
+        return buildMessage;
+    }
+
+    public void setBuildMessage(boolean buildMessage){
```
Comment (Manuri) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java]: Fix formatting

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java [hunk 3]
```diff
@@ -83,6 +85,14 @@ public void setAlgorithmClassName(String algorithmClassName) {
         this.algorithmClassName = algorithmClassName;
     }
 
+    public boolean isBuildMessage(){
```
Comment (Manuri) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java]: Fix formatting

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java [hunk 4]
```diff
@@ -83,6 +85,14 @@ public void setAlgorithmClassName(String algorithmClassName) {
...
+        return buildMessage;
+    }
+
+    public void setBuildMessage(boolean buildMessage){
```
Comment (Manuri) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java]: Fix formatting

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java [hunk 5]
```diff
@@ -92,7 +104,14 @@ public void build(OMElement elem, boolean isAnonymous) {
         if (isAnonymous) {
             elem.addAttribute("name", "anonymous", elem.getOMFactory().createOMNamespace("", ""));
         }
-        org.apache.synapse.endpoints.Endpoint failoverEndpoint = EndpointFactory.getEndpointFromElement(elem, isAnonymous, new Properties());
+        org.apache.synapse.endpoints.Endpoint failoverEndpoint = EndpointFactory
+                .getEndpointFromElement(elem, isAnonymous, new Properties());
+
+        if (failoverEndpoint instanceof org.apache.synapse.endpoints.FailoverEndpoint) {
```
Comment (pamod) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java]: Any reason for using fully qualified name here for endpoint definition ?

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java [hunk 6]
```diff
@@ -96,6 +106,10 @@ public OMElement serialize(OMElement parent) {
         OMElement loadbalance = fac.createOMElement("loadbalance", synNS);
         loadbalance.addAttribute(fac.createOMAttribute("algorithm", nullNS, algorithmClassName));
 
+        if (isBuildMessage()) {
+            loadbalance.addAttribute(fac.createOMAttribute("buildMessage", nullNS, Boolean.toString(true)));
```
Comment (pamod) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/loadbalance/LoadBalanceEndpoint.java]: shall we define the "buildMessage" as a constant ? could you also explain why we hardcode the value here ?

File: components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java (hunk 5 above)
Comment (Bhashinee) [inline components/mediation-ui/org.wso2.carbon.endpoint.ui/src/main/java/org/wso2/carbon/endpoint/ui/endpoints/failover/FailoverEndpoint.java]: This class also has the same name "FailoverEndpoint" as in  "org.apache.synapse.endpoints.FailoverEndpoint". So cannot import at the beginning.

**Comment by madhawa-gunasekara** [comment]
Merged.

**Pull request merged**